### DISCUSSION TITLE
Hoe kunnen gebruikers van de API de benodigde autorisatie verkrijgen?

### DISCUSSION THREAD

**Discussion by CelineJansen** [kadaster-labs/sensrnet-home#172] [Q&A] [answered]
Wanneer ik de API poog te gebruiken (https://demo.sensorenregister.nl/api/device bijvoorbeeld), krijg ik de response dat ik niet geautoriseerd ben (401, Unauthorized).

**Comment by kad-busses** [comment, marked as answer]
Om van de API gebruik te kunnen maken, dien je inderdaad ingelogd te zijn. Elke actie die via de API uitgevoerd kan worden is namelijk gekoppeld aan de ingelogde gebruiker.

Voor het verkrijgen van een authorisatietoken is een browser nodig en gaat het eenvoudigst op dezelfde manier als wanneer er in de frontend (https://demo.sensorenregister.nl/login) wordt ingelogd. Het token wordt dan in de browser opgeslagen en vervolgens meegestuurd met elk verzoek. 

De token kan dus verkregen worden door eerst zoals gebruikelijk in te loggen. Vervolgens kun je hem zien bij elk verzoek naar de backend onder de Network tab, of in de `registry-frontend-authzData` Session Storage onder de (Application) Storage tab. Beide zijn te vinden onder de browsers developer tools.

De token kan vervolgens gebruikt worden door aan de Bearer Authorization header toe te voegen. Via de standaard API interface (https://demo.sensorenregister.nl/api/) kan dat bijvoorbeeld door de token onder de "Authorize" knop toe te voegen. Daarna zijn alle endpoints beschikbaar.

**Reply by CelineJansen** [reply]
Bedankt voor de super uitgebreide reactie. Het is gelukt 👍
